### PR TITLE
Key component detailed documentation and agent file configuration documentation

### PR DESCRIPTION
**Pull request by ducpm511**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR is about these things:
- Restructuring the project
- Adding the detailed documentation for VCKit key components
- Adding the documentation for how to create/config the agent file
- Fixing a typo mistake
- Adding the tutorial's example source code.

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [vc-kit doc site](https://uncefact.github.io/vckit/)
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: documentation/docs/agent-configuration/config-agent-file.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  port: <CUSTOM PORT>
+  dbEncryptionKey: <SECRET KEY>
+  databaseFile: <DATA BASE FILE>
+  methods:
+  - <PLUGIN FUNCTIONS>
+```
+* `baseUrl`: This specifies the base URL for the application or service. 
+* `port`: This specifies the port number on which the application will listen for incoming connections. 
+* `dbEncryptionKey`: This field contains a cryptographic key used for encrypting data in the database. It's a hexadecimal string representing the encryption key. Run this command `veramo config gen-key` to generate a key.
+* `databaseFile`: This specifies the path to the SQLite database file.
```
Comment (ldhyen99) [inline documentation/docs/agent-configuration/config-agent-file.md:23]: Could I use other databases such as PostgreSQL or MongoDB? If I can, how can I configure it?

File: documentation/docs/get-started/demo-explorer-get-started/basic-operations.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ sidebar_position: 3
 # Basic Operations
 
 ## Create an indentifier
```
Comment (namhoang1604) [inline documentation/docs/get-started/demo-explorer-get-started/basic-operations.md:7]: Typo `indentifier`

File: documentation/docs/get-started/intro.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+sidebar_position: 1
+slug: /get-started
+---
+
+# Intro
+
+Let's discover **VCKit in less than 10 minutes**.
```
Comment (namhoang1604) [inline documentation/docs/get-started/intro.md:8]: Should the intro of get-started have the targets after reading through? Like
- Create VC
- Revoke VC
- Verify VC

File: documentation/docs/agent-configuration/config-agent-file.md
Comment (namhoang1604) [inline documentation/docs/agent-configuration/config-agent-file.md:1]: Before explaining the agent file, I think we need to introduce the syntax to let the developer understand it, like https://veramo.io/docs/veramo_agent/configuration_internals/#require-syntax

File: documentation/docs/agent-configuration/setup-library-agent.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                        },
+                        {
+                            chainId: 421613,
+                            name: 'arbitrum:goerli',
+                            rpcUrl:
+                                'https://arbitrum-goerli.infura.io/v3/' + INFURA_PROJECT_ID,
+                            registry: '0x8FFfcD6a85D29E9C33517aaf60b16FE4548f517E',
+                        },
+                    ],
+                }),
```
Comment (namhoang1604) [inline documentation/docs/agent-configuration/setup-library-agent.md:126]: Currently, the vckit supports only `did:web` and `did:key`, so we should remove the rest

**Review by namhoang1604 — CHANGES_REQUESTED** [review]

File: documentation/docs/get-started/demo-explorer-get-started/installation.md
Comment (ldhyen99) [inline documentation/docs/get-started/demo-explorer-get-started/installation.md:1]: Hi, when I click on `Make sure you have the API server started on local. See how to do it [here](/docs/api-server-get-started/installation).`, the page does not found.

`http://localhost:3001/docs/api-server-get-started/installation`

image

File: documentation/docs/agent-configuration/config-agent-file.md (hunk 1 above)
Comment (ducpm511) [reply documentation/docs/agent-configuration/config-agent-file.md:27]: you can use whatever type of database as long as it's supported by typeorm datasoure. I've updated the document to mention this.

File: documentation/docs/get-started/demo-explorer-get-started/basic-operations.md (hunk 2 above)
Comment (ducpm511) [reply documentation/docs/get-started/demo-explorer-get-started/basic-operations.md]: I fixed it

File: documentation/docs/get-started/intro.md (hunk 3 above)
Comment (ducpm511) [reply documentation/docs/get-started/intro.md:8]: I updated this

File: documentation/docs/get-started/demo-explorer-get-started/installation.md
Comment (ducpm511) [reply documentation/docs/get-started/demo-explorer-get-started/installation.md:1]: this one is fixed

File: documentation/docs/agent-configuration/config-agent-file.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+* `methods`: This specifies the functions that are exposed by the plugins integrated within the agent
+## Database connection
+### **`$require` syntax**
```
[the remark this answers is not in the document]
Comment (ducpm511) [reply documentation/docs/agent-configuration/config-agent-file.md]: I made this change

**Review by namhoang1604 — APPROVED** [review]

**Pull request merged**